File: saizad-mvvm/src/main/java/com/saizad/mvvm/components/form/ui/DOBStringInputFieldView.java
package com.saizad.mvvm.components.form.ui;

import android.content.Context;
import android.util.AttributeSet;

import com.saizad.mvvm.utils.Utils;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.joda.time.DateTime;

public class DOBStringInputFieldView extends InputFieldView<String> {

    public DOBStringInputFieldView(@NotNull Context context) {
        super(context);
    }

    public DOBStringInputFieldView(@NotNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public DOBStringInputFieldView(@NotNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public String resolveFrom(@NotNull CharSequence charSequence) {
        if(charSequence.toString().isEmpty()){
            return null;
        }
        return DateTime.parse(charSequence.toString()).toString(Utils.APP_DATE_FORMATTER);
    }

    @Override
    public CharSequence resolve(@Nullable String value) {
        if(value == null){
            return  null;
        }
        return DateTime.parse(value).toString(Utils.APP_DATE_FORMATTER);
    }

    @Override
    public boolean isSame(String value, String prevValue) {
        return value.equals(prevValue);
    }
}

File: saizad-mvvm/src/main/java/com/saizad/mvvm/components/form/ui/DOBStringInputFieldView.kt
package com.saizad.mvvm.components.form.ui

import android.content.Context
import android.util.AttributeSet
import com.saizad.mvvm.utils.Utils
import org.joda.time.DateTime

open class DOBStringInputFieldView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : InputFieldView<String>(context, attrs, defStyleAttr) {

    override fun resolveFrom(charSequence: CharSequence): String? {
        return if (charSequence.toString().isEmpty()) {
            null
        } else DateTime.parse(charSequence.toString())
            .toString(Utils.APP_DATE_FORMATTER)
    }

    override fun resolve(value: String?): CharSequence? {
        return if (value == null) {
            null
        } else DateTime.parse(value)
            .toString(Utils.APP_DATE_FORMATTER)
    }

    override fun isSame(value: String, prevValue: String): Boolean {
        return value == prevValue
    }
}